grab.py

```diff
@@ -39,64 +39,97 @@ def get_data(self):
 
 
 def strip_tags(html):
+    """Function to strip HTML tags, but preserve <pre> and <code> tags for code snippets."""
     s = MLStripper()
     s.feed(html)
-    return s.get_data()
+    
+    stripped_html = s.get_data()
+    
+    # Handle <pre> and <code> tags manually for code snippets
+    stripped_html = stripped_html.replace('<pre>', '\n.. code-block:: python\n\n    ')
+    stripped_html = stripped_html.replace('</pre>', '\n')
+    stripped_html = stripped_html.replace('<code>', '`')
+    stripped_html = stripped_html.replace('</code>', '`')
+    
+    return stripped_html
+
 
 def remove_indent(text):
+    """Removes leading and trailing spaces from each line in the text."""
     lines = text.splitlines()
     lines = [x.strip() for x in lines]
     return "\n".join(lines)
 
+
 def html2rst_allign_post(text):
-    lines = text.splitlines()
+    """Converts HTML content into reStructuredText (rst) format, handling code snippets."""
     if not text:
         raise ValueError("Empty post!")
 
-    if len(lines) < 2:
-        lines = '\n'.join(lines).replace("br>", "newline>\n").replace("/p>", "/pline>\n").splitlines()
-        lines = [x.replace("newline>", "br>").replace("/pline>","/p>") for x in lines]
+    lines = text.splitlines()
+    
+    # Insert teaser marker
     lines.insert(3, "<!-- TEASER_END -->")
+
+    # Convert <pre> and <code> HTML tags into rst code block formatting
+    text_with_code_blocks = text.replace('<pre>', '\n.. code-block:: python\n\n    ')
+    text_with_code_blocks = text_with_code_blocks.replace('</pre>', '\n')
+    text_with_code_blocks = text_with_code_blocks.replace('<code>', '`')
+    text_with_code_blocks = text_with_code_blocks.replace('</code>', '`')
+    
+    lines = text_with_code_blocks.splitlines()
     lines = ["    " + x.strip() for x in lines]
     lines = [".. raw:: html", ""] + lines
+    
     return "\n".join(lines)
 
+
 def grab_student(last_date, rss_url, project, student, season):
+    """Fetches blog posts of students and processes them into rst files."""
     feed = feedparser.parse(rss_url)
     dates = [last_date]
+    
     for item in feed['items']:
         item_date = dt.datetime.fromtimestamp(time.mktime(item['published_parsed']))
-        # item_date.tm_zone; tm_gmtoff
-        # wordpress; time.struct_time
+        
+        # Filter posts after the last grabbed date
         if item_date > last_date:
+            # Handle Medium posts specifically related to GSoC
             if "https://medium" in rss_url and not any('gsoc' in x.get('term').lower() for x in item.get('tags', [{'term': ''}])):
                 continue
+            
             print("#################### New post!")
             pp.pprint(item)
-            dates.append(item_date)
+            
```

Comment:
This probably shouldn't be deleted. Otherwise, it won't be able to return the max of the date at the end of the function.